napari/layers/utils/_tests/test_color_manager.py

```diff
@@ -133,6 +132,14 @@ def test_current_color_coercion(curr_color, expected):
     np.testing.assert_allclose(cm.current_color, expected)
 
 
+# TODO: should we change transform_color to handle an empty list
+# without a warning? Should the default be white or black like None?
```

Comment:
I don't love this idea! Where does the need come from to do anything at all with None or an empty list?

Reply:
In napari it comes from `transform_color`. We specifically handle [`None`](https://github.com/napari/napari/blob/19775aaf949fa0f27be37b56d7422d60556cd749/napari/utils/colormaps/standardize_color.py#L354). We specifically handle the case where we get an [empty array-like](https://github.com/napari/napari/blob/19775aaf949fa0f27be37b56d7422d60556cd749/napari/utils/colormaps/standardize_color.py#L202), though we do issue a warning.

The `None` behavior is consistent with `vispy.color.Color`, but the empty array-like raises an error in vispy.

```python
In [3]: from vispy.color import Color

In [4]: c = Color(None)

In [5]: c
Out[5]: <Color: (0.0, 0.0, 0.0, 0.0)>

In [6]: c = Color([])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-7-2099ae45e5f3> in <module>
----> 1 c = Color([])

~/software/miniconda3/envs/napari-dev/lib/python3.9/site-packages/vispy/color/color_array.py in __init__(self, color, alpha, clip)
    387         if isinstance(color, (list, tuple)):
    388             color = np.array(color, np.float32)
--> 389         rgba = _user_to_rgba(color, clip=clip)
    390         if rgba.shape[0] != 1:
    391             raise ValueError('color must be of correct shape')

~/software/miniconda3/envs/napari-dev/lib/python3.9/site-packages/vispy/color/color_array.py in _user_to_rgba(color, expand, clip)
     54     color = np.atleast_2d(color).astype(np.float32)
     55     if color.shape[1] not in (3, 4):
---> 56         raise ValueError('color must have three or four elements')
     57     if expand and color.shape[1] == 3:  # only expand if requested
     58         color = np.concatenate((color, np.ones((color.shape[0], 1))),

ValueError: color must have three or four elements
```

For `ColorManager.current_color` we have special handling of `None` and `[]` when in `DIRECT` mode, which is what the test being changed here is handling. On `main` we get opaque black instead of white with warning after this PR. As we might remove `ColorManager` in the nearish future, maybe we shouldn't change its behavior in this PR.

But we should decide what we going forward. I'm happy with coercing `None` to `[0, 0, 0, 0]` like vispy as it kind of represents no color, but also happy to not coerce it. I do not think we should coerce an empty array-like though, which just feels like a classic `ValueError` that should raise.